### PR TITLE
feature: add reinvent 2020 features

### PR DESCRIPTION
**Pull request by ChoiByungWook**
*Issue #, if available:*

*Description of changes:*
* Feature: Add Data Parallel support

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/container/1.6.0/Dockerfile.dlc.gpu [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+RUN rm -rf /root/.ssh/ && \
+ mkdir -p /root/.ssh/ && \
+ ssh-keygen -q -t rsa -N '' -f /root/.ssh/id_rsa && \
+ cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys \
+ && printf "Host *\n StrictHostKeyChecking no\n" >> /root/.ssh/config
+
+# TODO: Remove once the 1.6.0-gpu-py3 DLC image fixes MPI config
+# Comment line in MPI config to prevent mutually exclusive MCA settings
+RUN sed -i '62,62 s/^/#/' /home/.openmpi/etc/openmpi-mca-params.conf
```
Comment (ChaiBapchya) [inline test/container/1.6.0/Dockerfile.dlc.gpu:20]: https://github.com/aws/PRIVATE-deep-learning-containers/blob/4d4307b368dc10708556e73face8cd4efa2da4dd/pytorch/training/docker/1.6.0/py3/cu101/Dockerfile.gpu#L177
This was the line which was to be removed.

This was applicable for cu101.

However, reinvent DLC which was cu110 doesn't have that particular exclude line
https://github.com/aws/PRIVATE-deep-learning-containers/blob/private-master/pytorch/training/docker/1.6.0/py3/cu110/Dockerfile.gpu#L270-L278

Hence we don't need to run this commenting RUN statement anymore for cu110

Comment (ChoiByungWook) [reply test/container/1.6.0/Dockerfile.dlc.gpu:20]: It also looks like all of the other TODOs can be removed from the Dockerfile. Is that correct?

mpi4py: https://github.com/aws/PRIVATE-deep-learning-containers/blob/4d4307b368dc10708556e73face8cd4efa2da4dd/pytorch/training/docker/1.6.0/py3/cu101/Dockerfile.gpu#L152
OpenSSH: https://github.com/aws/PRIVATE-deep-learning-containers/blob/4d4307b368dc10708556e73face8cd4efa2da4dd/pytorch/training/docker/1.6.0/py3/cu101/Dockerfile.gpu#L181

File: test/container/1.6.0/Dockerfile.dlc.gpu [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# TODO: Remove once the 1.6.0-gpu-py3 DLC image fixes OpenSSH config
+# Configure OpenSSH so that nodes can communicate with each other
+RUN mkdir -p /var/run/sshd && \
+ sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+RUN rm -rf /root/.ssh/ && \
+ mkdir -p /root/.ssh/ && \
+ ssh-keygen -q -t rsa -N '' -f /root/.ssh/id_rsa && \
+ cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys \
+ && printf "Host *\n StrictHostKeyChecking no\n" >> /root/.ssh/config
```
Comment (ChaiBapchya) [inline test/container/1.6.0/Dockerfile.dlc.gpu:16]: https://github.com/aws/PRIVATE-deep-learning-containers/blob/4d4307b368dc10708556e73face8cd4efa2da4dd/pytorch/training/docker/1.6.0/py3/cu110/Dockerfile.gpu#L294-L298

Similarly we don't need this for cu11

Comment (ChoiByungWook) [reply test/container/1.6.0/Dockerfile.dlc.gpu:16]: See: https://github.com/aws/sagemaker-pytorch-training-toolkit/pull/225#discussion_r540585149

File: test/container/1.6.0/Dockerfile.dlc.gpu [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+# TODO: Remove once the 1.6.0-gpu-py3 DLC image fixes MPI config
+# Comment line in MPI config to prevent mutually exclusive MCA settings
+RUN sed -i '62,62 s/^/#/' /home/.openmpi/etc/openmpi-mca-params.conf
+
+COPY lib/changehostname.c /
+COPY lib/start_with_right_hostname.sh /usr/local/bin/start_with_right_hostname.sh
+RUN chmod +x /usr/local/bin/start_with_right_hostname.sh
```
Comment (ChaiBapchya) [inline test/container/1.6.0/Dockerfile.dlc.gpu:12]: https://github.com/aws/PRIVATE-deep-learning-containers/blob/4d4307b368dc10708556e73face8cd4efa2da4dd/pytorch/training/docker/1.6.0/py3/cu110/Dockerfile.gpu#L133
similarly

File: test/container/1.6.0/Dockerfile.dlc.gpu (hunk 1 above)
Comment (ChaiBapchya) [reply test/container/1.6.0/Dockerfile.dlc.gpu]: Yes please!

File: test/integration/sagemaker/test_smdataparallel.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+):
+    default_bucket = sagemaker_session.default_bucket()
+    output_path = "s3://" + os.path.join(default_bucket, "pytorch/smdataparallel")
+
+    estimator = PyTorch(
+        entry_point=os.path.join(resources_path, "mnist", "smdataparallel_mnist.py"),
+        role="SageMakerRole",
+        train_instance_type=train_instance_type,
+        sagemaker_session=sagemaker_session,
+        train_instance_count=instances,
```
Comment (ChaiBapchya) [inline test/integration/sagemaker/test_smdataparallel.py:41]: Why not SM SDK 2.x format?

**Review by ChaiBapchya — APPROVED** [review]
Thanks! LGTM!

**Review by metrizable — APPROVED** [review]

**Pull request merged**